tests/utils/test_maven.py

```diff
@@ -0,0 +1,119 @@
+
+    def test_generate_maven_servers_exist(self):
+        maven_servers = [
+            {
+                "id": "one",
+                "username": "user1",
+                "password": "password1"
+            }
+        ]
+
+        settings = '''<settings><servers><server><id>one</id><username>user1</username><password>password1</password></server></servers></settings>'''
+
+        with TempDirectory() as temp_dir:
+            generate_maven_settings(temp_dir.path, maven_servers, None, None)
+            with open(temp_dir.path + '/settings.xml', 'r') as tester:
+                results = tester.read()
+                assert results == settings
+
+    def test_generate_maven_servers_id_does_not_exist(self):
+        maven_servers = [
+            {
+                "username": "user1",
+                "password": "password1"
+            }
+        ]
+        with TempDirectory() as temp_dir:
+            with self.assertRaisesRegex(ValueError, 'id is required for maven_servers.'):
+                generate_maven_settings(temp_dir.path, maven_servers, None, None)
+
+    def test_generate_maven_servers_username_does_not_exist(self):
+        maven_servers = [
+            {
+                "id": "one",
+                "password": "password1"
+            }
+        ]
+        with TempDirectory() as temp_dir:
+            with self.assertRaisesRegex(ValueError,
+                                        'username and password are required for maven_servers.'):
+                generate_maven_settings(temp_dir.path, maven_servers, None, None)
+
+    def test_generate_maven_repositories_exists(self):
+        maven_repositories = [
+            {
+                "id": "repo1",
+                "url": "repo_url1",
+                "releases": "true",
+                "snapshots": "true"
+            }
+        ]
+        settings = '''<settings><profiles><profile><repositories><repository><id>repo1</id><url>repo_url1</url><releases><enabled>true</enabled></releases><snapshots><enabled>true</enabled></snapshots></repository></repositories></profile></profiles></settings>'''
+
+        with TempDirectory() as temp_dir:
+            generate_maven_settings(temp_dir.path, None, maven_repositories, None)
+            with open(temp_dir.path + '/settings.xml', 'r') as tester:
+                results = tester.read()
+                assert results == settings
+
+    def test_generate_maven_repositories_id_does_not_exists(self):
+        maven_repositories = [
+            {
+                "url": "repo_url1",
+                "releases": "true",
+                "snapshots": "true"
+            }
+        ]
+        with TempDirectory() as temp_dir:
+            with self.assertRaisesRegex(ValueError,
+                                        'id and url are required for maven_repositories.'):
+                generate_maven_settings(temp_dir.path, None, maven_repositories, None)
+
+    def test_generate_maven_mirrors_exists(self):
```

Comment:
Do you have a test for when mirrors are empty?

Reply:
@akim309 - passing to you

Reply:
@akim309  Your tests currently cover the step properly. You can ignore this request.